internal/controller/providers/pg/provider.go

```diff
@@ -147,6 +149,32 @@ func (p *Provider) Status(ctx context.Context) (everestv1alpha1.DatabaseClusterS
 		status.Status = everestv1alpha1.AppStateRestoring
 	}
 
+	// If the PVC resize is currently in progress, or just finished, we need to
+	// check if it failed in order to set or clear the error condition.
+	if status.Status == everestv1alpha1.AppStateResizingVolumes ||
+		prevStatus.Status == everestv1alpha1.AppStateResizingVolumes {
+		meta.RemoveStatusCondition(&status.Conditions, everestv1alpha1.ConditionTypeVolumeResizeFailed)
+		if failed, condMessage, err := common.VerifyPVCResizeFailure(ctx, p.C, p.DB.GetName(), p.DB.GetNamespace()); err != nil {
+			return status, err
+		} else if failed {
+			status.Status = everestv1alpha1.AppStateResizingVolumes
+			meta.SetStatusCondition(&status.Conditions, metav1.Condition{
+				Type:               everestv1alpha1.ConditionTypeVolumeResizeFailed,
+				Status:             metav1.ConditionTrue,
+				Reason:             everestv1alpha1.ReasonVolumeResizeFailed,
+				Message:            condMessage,
+				LastTransitionTime: metav1.Now(),
+				ObservedGeneration: p.DB.GetGeneration(),
+			})
+		}
+	}
```

Comment:
These conditions are identical for all 3 db types, would we want to have it in the common code?

Reply:
Right, we can leave it as it is for readability, just wanted to bring it to consideration. 
There is always this dilemma - now it looks similar however there is no guarantee we won't need to handle some specifics for the particular engines... 
Feel free to resolve the comment.